stanford/roles/cluster/tasks/instance.yml

```diff
@@ -25,38 +25,31 @@
 - debug:
     var: item
   with_items: "{{ ec2.instances | default([])}}"
-- name: Add new instance to host group
-  add_host:
-    name: "{{ item.public_ip }}"
-    groups:
-      - cluster_group_launched
-      - tag_todo_deploy
-      - "tag_Deployment_{{ COMMON_DEPLOYMENT }}"
-      - "tag_Cluster_{{ CLUSTER_NAME }}"
-      - "tag_Number_{{ CLUSTER_NUMBER }}"
+- name: Set ansible host facts
+  set_fact:
+    cluster_ansible_host: "{{ item.private_ip }}"
+  with_items: "{{ ec2.instances | default([])}}"
+- name: Set ansible host facts
+  set_fact:
+    cluster_ansible_host: "{{ item.public_ip }}"
   with_items: "{{ ec2.instances | default([])}}"
-  when:
-    - item.public_ip
+  when: item.public_ip
 - name: Add new instance to host group
   add_host:
-    name: "{{ item.private_ip }}"
+    name: "{{ cluster_ansible_host }}"
     groups:
       - cluster_group_launched
       - tag_todo_deploy
       - "tag_Deployment_{{ COMMON_DEPLOYMENT }}"
       - "tag_Cluster_{{ CLUSTER_NAME }}"
       - "tag_Number_{{ CLUSTER_NUMBER }}"
-  with_items: "{{ ec2.instances | default([])}}"
-  when:
-    - not item.public_ip
+      - "tag_Name_{{ CLUSTER_INSTANCE_TAG_NAME }}"
 - name: Wait for SSH to come up
   wait_for:
-    host: "{{ item.public_ip | default(item.private_ip) }}"
+    host: "{{ cluster_ansible_host }}"
     port: 22
     delay: 20
     timeout: 120
-  with_items: "{{ ec2.instances | default([])}}"
-  ignore_errors: yes
```

Comment:
This used to unconditionally fail, because even though `item.public_ip` didn't actually contain a public IP address, it was at least _defined_, which meant the fallback `item.private_ip` was never used.
This meant this task would wait the full 2 minutes, never able to connect to the server (unless deploying the jumpbox or other public instance), ignored the error, then continued.

Now, since it uses the (correct) address assigned above, this task actually does succeed in under 30 seconds. That could probably even be decreased, if the `delay` value above were decreased, but I'll save that for later.